metrics/ci_integrations/test/source/github_actions/adapter/github_actions_source_client_adapter_test.dart

```diff
@@ -0,0 +1,1022 @@
+          lastBuild,
+        );
+
+        expect(result, completion(isEmpty));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() fetches coverage for each build",
+      () async {
+        final runsPage = WorkflowRunsPage(
+          values: testDataGenerator.generateWorkflowRunsByNumbers(
+            runNumbers: [4, 3, 2, 1],
+          ),
+        );
+
+        whenFetchWorkflowRuns(
+          withArtifactsPage: defaultArtifactsPage,
+          withJobsPage: defaultJobsPage,
+        ).thenSuccessWith(runsPage);
+
+        final lastBuild = testDataGenerator.generateBuildData(buildNumber: 1);
+
+        final expected = [
+          testDataGenerator.coverage,
+          testDataGenerator.coverage,
+          testDataGenerator.coverage,
+        ];
+
+        final result = await adapter.fetchBuildsAfter(
+          testDataGenerator.jobName,
+          lastBuild,
+        );
+
+        final coverage = result.map((build) => build.coverage).toList();
+
+        expect(coverage, equals(expected));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() fetches coverage for each build using pagination for run artifacts",
+      () async {
+        final runsPage = WorkflowRunsPage(
+          values: testDataGenerator.generateWorkflowRunsByNumbers(
+            runNumbers: [4, 3, 2, 1],
+          ),
+        );
+
+        whenFetchWorkflowRuns(
+          withArtifactsPage: emptyArtifactsPage,
+          withJobsPage: defaultJobsPage,
+        ).thenSuccessWith(runsPage);
+
+        when(githubActionsClientMock.fetchRunArtifactsNext(emptyArtifactsPage))
+            .thenSuccessWith(defaultArtifactsPage);
+
+        final lastBuild = testDataGenerator.generateBuildData(buildNumber: 1);
+
+        final expectedCoverage = [
+          testDataGenerator.coverage,
+          testDataGenerator.coverage,
+          testDataGenerator.coverage,
+        ];
+
+        final result = await adapter.fetchBuildsAfter(
+          testDataGenerator.jobName,
+          lastBuild,
+        );
+
+        final coverage = result.map((build) => build.coverage).toList();
+
+        expect(coverage, equals(expectedCoverage));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() does not fetch the skipped builds",
+      () {
+        final workflowRunJobsPage = WorkflowRunJobsPage(values: const [
+          WorkflowRunJob(conclusion: GithubActionConclusion.skipped)
+        ]);
+
+        final lastBuild = testDataGenerator.generateBuildData(buildNumber: 1);
+
+        whenFetchWorkflowRuns(
+          withArtifactsPage: defaultArtifactsPage,
+          withJobsPage: workflowRunJobsPage,
+        ).thenSuccessWith(defaultRunsPage);
+
+        final result = adapter.fetchBuildsAfter(
+          testDataGenerator.jobName,
+          lastBuild,
+        );
+
+        expect(result, completion(isEmpty));
+      },
+    );
+
+    test(
+      ".fetchBuildsAfter() fetches builds using pagination for workflow runs",
```

Comment:
Should we add the following test case: 
- .fetchBuildsAfter() fetches builds with a greater run number than the given if the given number is not found

?